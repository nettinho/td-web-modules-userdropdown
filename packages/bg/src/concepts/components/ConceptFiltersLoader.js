import React from "react";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";
import { bindActionCreators, compose } from "redux";
import { connect } from "react-redux";
import { Loading } from "@truedat/core/components";
import { clearConceptFilters, fetchConceptFilters } from "../routines";

export class ConceptFiltersLoader extends React.Component {
  static propTypes = {
    clearConceptFilters: PropTypes.func,
    fetchConceptFilters: PropTypes.func,
    conceptFiltersLoading: PropTypes.bool
  };

  componentDidMount() {
    const { fetchConceptFilters } = this.props;
    fetchConceptFilters();
  }

  componentWillUnmount() {
    const { clearConceptFilters } = this.props;
    clearConceptFilters();
  }

  render() {
    const { conceptFiltersLoading } = this.props;

    if (conceptFiltersLoading) {
      return <Loading />;
    } else {
      return null;
    }
  }
}

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ clearConceptFilters, fetchConceptFilters }, dispatch)
});

const mapStateToProps = ({ conceptFiltersLoading }) => ({
  conceptFiltersLoading
});

export default compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(ConceptFiltersLoader);
