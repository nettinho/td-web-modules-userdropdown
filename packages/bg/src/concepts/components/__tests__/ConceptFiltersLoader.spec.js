import React from "react";
import { shallow } from "enzyme";
import { ConceptFiltersLoader } from "../ConceptFiltersLoader";

describe("<ConceptFiltersLoader />", () => {
  const conceptFiltersLoading = false;
  const fetchConceptFilters = jest.fn();
  const clearConceptFilters = jest.fn();

  it("matches the latest snapshot", () => {
    const conceptFiltersLoading = true;
    const props = {
      clearConceptFilters,
      fetchConceptFilters,
      conceptFiltersLoading
    };
    const wrapper = shallow(<ConceptFiltersLoader {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a loader if conceptFiltersLoading is true", () => {
    const conceptFiltersLoading = true;
    const props = {
      clearConceptFilters,
      fetchConceptFilters,
      conceptFiltersLoading
    };
    const wrapper = shallow(<ConceptFiltersLoader {...props} />);
    expect(wrapper.find("Loading").length).toBe(1);
  });

  it("renders null if conceptFiltersLoading is false", () => {
    const props = {
      clearConceptFilters,
      fetchConceptFilters,
      conceptFiltersLoading
    };
    const wrapper = shallow(<ConceptFiltersLoader {...props} />);
    expect(wrapper.getElement()).toBeNull();
  });

  it("calls fetchConceptFilters when component mounts but not when it unmounts", () => {
    const fetchConceptFilters = jest.fn();
    const props = {
      clearConceptFilters,
      fetchConceptFilters,
      conceptFiltersLoading
    };
    jest.spyOn(ConceptFiltersLoader.prototype, "componentDidMount");
    const wrapper = shallow(<ConceptFiltersLoader {...props} />);
    expect(
      ConceptFiltersLoader.prototype.componentDidMount.mock.calls.length
    ).toBe(1);
    expect(fetchConceptFilters.mock.calls.length).toBe(1);
    wrapper.unmount();
    expect(fetchConceptFilters.mock.calls.length).toBe(1);
  });

  it("calls clearConceptFilters when component unmounts but not when it mounts", () => {
    const clearConceptFilters = jest.fn();
    const props = {
      clearConceptFilters,
      fetchConceptFilters,
      conceptFiltersLoading
    };
    jest.spyOn(ConceptFiltersLoader.prototype, "componentWillUnmount");
    const wrapper = shallow(<ConceptFiltersLoader {...props} />);
    expect(clearConceptFilters.mock.calls.length).toBe(0);
    wrapper.unmount();
    expect(
      ConceptFiltersLoader.prototype.componentWillUnmount.mock.calls.length
    ).toBe(1);
    expect(clearConceptFilters.mock.calls.length).toBe(1);
  });
});
