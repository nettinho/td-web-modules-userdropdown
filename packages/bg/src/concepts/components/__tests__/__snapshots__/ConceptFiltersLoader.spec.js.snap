// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConceptFiltersLoader /> matches the latest snapshot 1`] = `<Loading />`;
